knative-operator/pkg/common/monitoring.go

```diff
@@ -0,0 +1,136 @@
+package common
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
+	"os"
+
+	mfclient "github.com/manifestival/controller-runtime-client"
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// installedNamespaceEnvKey is the ns where Openshift serverless operator has been installed
+	installedNamespaceEnvKey = "NAMESPACE"
+	// operatorDeploymentNameEnvKey is the name of the deployment of the Openshift serverless operator
+	operatorDeploymentNameEnvKey = "DEPLOYMENT_NAME"
+	// service monitor created successfully when monitoringLabel added to namespace
+	monitoringLabel = "openshift.io/cluster-monitoring"
+	rolePath        = "deploy/role_service_monitor.yaml"
+	testRolePath    = "TEST_ROLE_PATH"
+)
+
+func SetupMonitoringRequirements(api client.Client) error {
+	ns, err := getOperatorNamespace()
+	if err != nil {
+		return err
+	}
+	err = addMonitoringLabelToNamespace(ns, api)
```

Comment:
The monitoring label can automatically get added to our namespace with recent OLM integration for monitoring with the console. The same goes for the appropriate roles. Are we sure we want to handle these separately in our operator instead of letting OLM do this on our behalf?

Reply:
Please read my comment bellow.